### ISSUE TITLE
Updated: googletag.display(...) doesn't work properly in Firefox

### ISSUE THREAD

**Issue by kosik86** [krux/postscribe#86]
Hey 

I recently noticed very strange behavior of postscribe. It seems to be Firefox specific, and hard to debug.
I prepared set of demo pages.
All of these demos work properly in Chrome and there are some issues in Firefox.

Example 1 loads the ads after clicking on the button, instead of loading them immediately.
It works if done without postscribe, it doesn't with postscribe (only in Firefox).

Example 2 loads the ad immediately using postscribe, what works fine in all browsers, even Firefox.

The ad loading implementation has two steps:
1. Loading main script with all ad slots definitions and with googletag main script
2. Loading single ad slots scripts, they are rather simply - googletag.display() call mainly

See the example:
http://jsfiddle.net/5xunxpmL/

This one calls postscribe immediately (exactly the same method like previously, but
without any timeout):
http://jsfiddle.net/5xunxpmL/1/

Could you please check if you can observe the same issue in Firefox?
The actual ad request is getting aborted in Firefox for some reason, and the creative is never fetched.
GET ads?gdfp_req=1&correlato...588686&ga_hid=772171976 
Aborted

I can't see anything wrong with the code.
I would appreciate any help.

**Comment by jnewman** [comment]
This one is pretty odd. I noticed that if I click the load w/ postscribe twice it also works. I'll check it out on some older browsers and see where we land.

**Comment by kosik86** [comment]
Thank you for confirming. It would be really great to see any fix or workaround for the bug.

**Comment by zulkamal** [comment]
I'm having the same issue as well and here's what I found.

I am unable to use GPT's async mode and can confirm changing to async mode causes the issue to disappear.

If you put a breakpoint at line 752: `_this.onScriptDone(tok);` this is what happens at run/refresh in chrome:

1. `tok` content property is the `googletag.display('"div-gpt-ad-1416362643484"')`
2. Continuing execution displays the creative and calls it a day.

However in firefox - also at first run/refresh: 
1. Creative is displayed first without hitting the breakpoint.
2. Breakpoint is hit. `tok` content value is now:

```javascript
<script>googletag.impl.pubads.createDomIframe("div-gpt-ad-1416362643484-0_ad_container","/1234/abcd/700x350_0",false,undefined);</script>
```
3. Continuing execution causes the creative to disappear. 
4. Breakpoint is hit again. `tok` content value is now the initial script: `googletag.display('"div-gpt-ad-1416362643484"')`
5. Page is fully loaded

To my knowledge, if there are any DOM changes to the div the `display()` call is in - like moving it to another parent) - the creative's iframe clears out and a yellow fill replaces it. Item 3 in the firefox sequence seems to be exhibiting that behavior. 

I hope this sheds some light on the issue.

**Comment by carambolage** [comment]
I've debugged it like @zulkamal 
Both issues with the Google Publisher Tags discussed here and in #75 depend on the same behaviour of the GPTs. Finally the following function is called:
```
googletag.impl.pubads.createDomIframe
```
I have committed an workaorund that uses postscribe's normal workflow: 3ca378fedf008351d1ed2da9ebfacc06b0eb26fa
The google function is removed by clearing the token's content before manipulating the DOM.
```
// Replace cursor with style.
cursor.parentNode.replaceChild(el, cursor);
```
Instead, the evaluation is stacked on the writeQueue
```
// ### Script tokens
WriteStream.prototype.handleScriptToken = function(tok) {
  if (/googletag.impl.pubads.createDomIframe/.test(tok.content)) {
	var domiframe = tok.content;
	tok.length -= tok.content.length;
	tok.content = "";
	this.writeQueue.unshift(function createDomIframe() {
		eval(domiframe);
	});
  }
```
http://jsfiddle.net/4y37qk8h/

**Comment by carambolage** [comment]
Hm, i forgot to escape the dot in the regular expression...

I thought about improving this workaround.
I think you can avoid using ```eval```, if you create an regular expression that matches the call of the function ```googletag.impl.pubads.createDomIframe``` with its parameters.
I suppose, the token's' content theoretically could contain more than the function call.
With the regular expression you could remove the function call and keep the rest.

What do you basically think about this workaorund?
Is there a regular expression guru that knows the right expression?

I don't know the function signature. In my example ```googletag.impl.pubads.createDomIframe("ledaerboard_1_ad_container" ,"/5919292/test/postscribe_0",false,false);``` is called.
I propose to take care of undefined amount and types of parameters.

**Comment by sethyates** [comment]
We believe this is fixed in 2.0.0.  Please retest against 2.0.0 release and open a new issue following Issue Reporting guidelines in 2.0.0 if you believe this is still an issue.
